### ISSUE TITLE
`implicit_return` should not trigger when there are closure scoped local functions following

### ISSUE THREAD

**Issue by MartinP7r** [realm/SwiftLint#5161]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

```swift
func foo() {
    return 1
    func bar() {}
}
```
This will trigger `implicit_return`, even though the `return` is necessary.
